### PR TITLE
Feat/car for rent

### PR DESCRIPTION
**Pull request by tringuyen-dev**

### CODE REVIEW

**Comment by tinhlesc** [comment]
@tringuyennfq please refer how to put the name of folder from Khoa Tran.
You should put the name like: Controller, Bootstrap, Database or something like that.

File: src/view/login.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
```
Comment (bangdinhnfq) [inline src/view/login.php]: Don't need. Should remove.

Comment (bangdinhnfq) [inline src/view/login.php]: Don't need. Should remove.

File: src/view/login.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+               autofocus>
+        <label for="inputPassword" class="sr-only">Password</label>
+        <input type="password" id="inputPassword" class="form-control" name="password" placeholder="Password" required>
+        <div class="checkbox mb-3">
+            <label>
+                <input type="checkbox" value="remember-me"> Remember me
+            </label>
+        </div>
+        <button class="btn btn-lg btn-primary btn-block" type="submit">Sign in</button>
+        <p class="mt-5 mb-3 text-muted">&copy; 2022 <?php echo $name; ?></p>
```
Comment (bangdinhnfq) [inline src/view/login.php]: It should be 
```
<?= echo $name; ?>
```

Comment (bangdinhnfq) [inline src/view/login.php]: It should be 
```
<?= echo $name; ?>
```

File: src/bootstrap/View.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+namespace Tringuyen\CarForRent\bootstrap;
+
+class View
+{
+
+    public static function renderView(string $view, $params = [])
+    {
+        $layoutContent = self::layoutContent();
+        $viewContent = self::viewContent($view, $params);
+        return str_replace('{{content}}', $viewContent, $layoutContent);
```
Comment (bangdinhnfq) [inline src/bootstrap/View.php]: content must be variable

Comment (bangdinhnfq) [inline src/bootstrap/View.php]: content must be variable

File: src/bootstrap/View.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+
+    }
+
+    protected static function layoutContent()
+    {
+        ob_start();
+        include_once Application::$ROOT_DIR . "/src/view/layouts/main.php";
```
Comment (bangdinhnfq) [inline src/bootstrap/View.php]: Should define as a constant

Comment (bangdinhnfq) [inline src/bootstrap/View.php]: Should define as a constant

File: src/bootstrap/View.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        return ob_get_clean();
+    }
+
+    protected static function viewContent($view, $params)
+    {
+        foreach ($params as $key => $value) {
+            $$key = $value;
+        }
+        ob_start();
+        include_once Application::$ROOT_DIR . "/src/view/$view.php";
```
Comment (bangdinhnfq) [inline src/bootstrap/View.php]: Should define as a constant

Comment (bangdinhnfq) [inline src/bootstrap/View.php]: Should define as a constant

File: src/database/DatabaseConnect.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        if (!self::$conn) {
+            $host = 'localhost';
+            $username = 'carrent';
+            $password = 'Nfq@2022';
+            $database = 'rentcar';
+
+            try {
+                self::$conn = new PDO("mysql:host=$host;dbname=$database", $username, $password);
+                self::$conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+                echo "Connected successfully";
```
Comment (bangdinhnfq) [inline src/database/DatabaseConnect.php]: Remove echo

Comment (bangdinhnfq) [inline src/database/DatabaseConnect.php]: Remove echo

File: src/database/DatabaseConnect.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+            } catch (PDOException $e) {
+                echo "Connection failed: " . $e->getMessage();
```
Comment (bangdinhnfq) [inline src/database/DatabaseConnect.php]: Should handle exception

Comment (bangdinhnfq) [inline src/database/DatabaseConnect.php]: Should handle exception

File: public/index.php [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+use Tringuyen\CarForRent\bootstrap\Application;
+use Tringuyen\CarForRent\controller\SiteController;
+use Tringuyen\CarForRent\database\DatabaseConnect;
+
+require_once __DIR__.'/../vendor/autoload.php';
+
+error_reporting(E_ALL);
+ini_set('display_errors', '1');
```
Comment (bangdinhnfq) [inline public/index.php]: Remove debug code

Comment (bangdinhnfq) [inline public/index.php]: Remove debug code

**Review by bangdinhnfq — CHANGES_REQUESTED** [review]

File: Tests/Tringuyen/CarForRent/Model/UserTest.php [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+     * @return void
+     */
+    public function testGetPassword(): void
+    {
+        $user1 = new User();
+        $user1->setPassword('123456');
+        $result = $user1->getPassword();
+        $this->assertEquals('123456', $result);
+    }
+}
```
Comment (bangdinhnfq) [inline Tests/Tringuyen/CarForRent/Model/UserTest.php:42]: EOL

File: src/Controller/UserLoginController.php [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+//             return View::renderView('User/login', [
+//            'title' => 'Login',
+//            'username' => $_POST['username'] ?? '',
+//            'password' => $_POST['password'] ?? '',
+//            'error' => ''
+//        ]);
+
+    }
+
+//    public function handleLogin()
```
Comment (bangdinhnfq) [inline src/Controller/UserLoginController.php]: Remove unused code

File: src/Exception/LoginException.php [hunk 11]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Tringuyen\CarForRent\Exception;
+
+use Exception;
+
+class LoginException extends Exception
+{
+
+}
```
Comment (bangdinhnfq) [inline src/Exception/LoginException.php]: EOL

**Review by bangdinhnfq — CHANGES_REQUESTED** [review]

**Pull request merged**